From 56326e463a56afa0e632baad4c94faa7aa154486 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Fri, 20 Feb 2015 17:33:56 +0100
Subject: [PATCH] metadata: return if meta_tree_refresh failed

G_UNLOCK is called twice if meta_tree_refresh fails and also concurrent
access is possible on cached_trees. However this isn't problem currently,
because meta_tree_lookup_by_name is always called with for_write=FALSE
(except for meta-set testing utility), so this if statement is never issued.

https://bugzilla.gnome.org/show_bug.cgi?id=598561
---
 metadata/metatree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metadata/metatree.c b/metadata/metatree.c
index 2983e5d..0fe29a1 100644
--- a/metadata/metatree.c
+++ b/metadata/metatree.c
@@ -547,7 +547,7 @@ meta_tree_lookup_by_name (const char *name,
         return tree;
 
       meta_tree_unref (tree);
-      tree = NULL;
+      return NULL;
     }
 
   filename = g_build_filename (g_get_user_data_dir (), "gvfs-metadata", name, NULL);
-- 
2.5.0

